src/game.ts

```diff
@@ -0,0 +1,122 @@
+import Board from './board';
+import Gameplay from './gameplay';
+import Card from './card';
+import { imageNames } from './imageNames';
+
+export default class Game {
+    private board: Board;
+    private gameplay: Gameplay;
+    private imagesCount = imageNames.length;
+    private selectedImagesIds: number[] = [];
+    private cardEvent = this.manageCardsVisibility.bind(this);
+    public cards: Card[] = [];
+
+    public startGame(): void {
+        this.board = new Board();
+        this.board.createBoard();
+        this.gameplay = new Gameplay(this.cards);
+        this.selectRandomImages();
+        this.createCards();
+        this.addClickEventToCards();
+    }
+
+    public newGame(): void {
+        this.gameplay.playerMoves = 0;
+        document.querySelector('.board').innerHTML = '';
+        this.board.renderEmptyCards();
+        this.selectRandomImages();
+        this.cards = [];
+        this.createCards();
+        this.gameplay = new Gameplay(this.cards);
+        this.addClickEventToCards();
+    }
+
+    private selectRandomImages(): void {
+        const uniqueCardsCount: number = this.board.cardCount / 2;
+        let currentNumber: number = 0;
+
+        for (let x = 0; this.selectedImagesIds.length < uniqueCardsCount; x++) {
+            currentNumber = Math.floor(Math.random() * this.imagesCount);
+            if (!this.selectedImagesIds.includes(currentNumber)) {
+                this.selectedImagesIds.push(currentNumber);
+            }
+        }
+    }
+
+    private createCards() {
+        let currentNumber: number = 0;
+        let allIdsTwice: number[] = [...this.selectedImagesIds, ...this.selectedImagesIds];
+        let idCounter: number = 0;
+
+        for (let x = 0; this.cards.length < this.board.cardCount; x++) {
+            currentNumber = Math.floor(Math.random() * allIdsTwice.length);
+
+            if (this.cardOccurredLessThanTwice(imageNames[allIdsTwice[currentNumber]])) {
+                this.cards.push(
+                    new Card(imageNames[allIdsTwice[currentNumber]], idCounter)
+                );
+                allIdsTwice.splice(currentNumber, 1);
+                idCounter++;
+            }
+        }
+    }
```

Comment:
To jest element przygotowania elementów wejściowych do rozgrywki, niekoniecznie samej rozgrywki.

Wyciągnął bym to do jakiejś osobnej klasy, może być nawet statyczna, choć wolałbym żeby była normlana i wstrzyknięta przez konstruktor.